Sources/Data/models/PlaylistFolder.swift

```diff
@@ -57,10 +70,55 @@ final public class PlaylistFolder: NSManagedObject, CRUD, Identifiable {
       playlistFolder.dateAdded = Date()
       playlistFolder.order = Int32.min
       playlistFolder.uuid = folderId
+      playlistFolder.sharedFolderId = nil
 
       PlaylistFolder.reorderItems(context: context)
       PlaylistFolder.saveContext(context)
-
+      
+      DispatchQueue.main.async {
+        completion?(folderId)
+      }
+    }
+  }
+  
+  public static func addInMemoryFolder(title: String, creatorName: String, creatorLink: String, sharedFolderId: String, completion: ((_ folder: PlaylistFolder, _ uuid: String) -> Void)? = nil) {
+    DataController.perform(context: .existing(DataController.viewContextInMemory), save: false) { context in
+      context.perform {
+        let folderId = UUID().uuidString
+        let playlistFolder = PlaylistFolder(context: context)
+        playlistFolder.title = title
+        playlistFolder.creatorName = creatorName
+        playlistFolder.creatorLink = creatorLink
+        playlistFolder.dateAdded = Date()
+        playlistFolder.order = Int32.min
+        playlistFolder.uuid = folderId
+        playlistFolder.sharedFolderId = sharedFolderId
+        
+        PlaylistFolder.reorderItems(context: context)
+        PlaylistFolder.saveContext(context)
+        
+        DispatchQueue.main.async {
+          completion?(playlistFolder, folderId)
+        }
+      }
+    }
+  }
+  
+  public static func saveInMemoryFolderToDisk(folder: PlaylistFolder, completion: ((_ uuid: String) -> Void)? = nil) {
+    DataController.perform(context: .existing(DataController.viewContext), save: false) { context in
+      let folderId = folder.uuid ?? UUID().uuidString
+      let playlistFolder = PlaylistFolder(context: context)
+      playlistFolder.uuid = folder.uuid
+      playlistFolder.title = folder.title
+      playlistFolder.order = Int32.min
```

Comment:
Shouldnt it be `folder.order`

Reply:
No. The folder order isn't guaranteed because Swift's JSON serialization is un-ordered.
So by setting it as `Int32.min`, then calling `Re-order` on the next few lines, it will re-order it properly.